lib/Platform/ElasticSearchPlatform.php

```diff
@@ -409,6 +414,29 @@ private function connectToElastic(array $hosts) {
 		}
 	}
 
+	/**
+	 * @param array $hosts
+	 */
+	private function processHostUrls(array $hosts) {
+		$user = null;
+		$pass = null;
+		$cleanedHosts = [];
+
+		foreach ($hosts as $host) {
+			$parts = parse_url($host);
+			if (!$user && isset($parts['user']) && isset($parts['pass'])) {
+				$user = $parts['user'];
+				$pass = $parts['pass'];
+			}
+			$cleanedHosts[] = $this->cleanHost($host);
+		}
+
+		return [
+			'hosts' => $cleanedHosts,
+			'user' => $user,
+			'pass' => $pass
+		];
```

Comment:
And then you could change all of this to something like this:

```php
// Note: we assume that the user/pass combination is the same in all the hosts because ElasticSearch requires all hosts to have the same user/pass
$user = parse_url($hosts[0], PHP_URL_USER);
$pass = parse_url($hosts[0], PHP_URL_PASS);
return compact('user', 'pass');
```

You could even inline this function because it's very short now.